docs: Add programmatic component input/output/directive binding

SkyZeroZx opened the pull request:
Expands the guide to explain how inputs, outputs, and directives can be provided when creating components programmatically

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+
+## Providing inputs, outputs, and directives at creation
+
+The `ViewContainerRef.createComponent` API supports binding inputs/outputs and attaching directives in one step. Instead of manually setting inputs and subscribing to outputs after creation, you can pass a `bindings` array with helpers like `inputBinding()`, `outputBinding()`, and `twoWayBinding()`, as well as a `directives` array for any directives to apply. This allows you to programmatically i ...
```
With probably should mention both `createComponent` and `ViewContainerRef.createComponent` in that intro paragraph

JeanMeche commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+import { ThemeDirective } from '../theme.directive';
+
+@Component({
+  template: `<ng-container #container></ng-container>`
```
```suggestion
  template: `<ng-container #container />`
```

SkyZeroZx replied:
Done

mauriziocescon commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+    private injector: EnvironmentInjector,
+    private appRef: ApplicationRef,
+  ) {}
+
```
maybe `inject` should be used here

JeanMeche commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+      ],
+    });
+
+    // Attach to change detection and add to DOM
+    this.appRef.attachView(ref.hostView);
+    document.body.appendChild(host);
```
This part probably needs more explanation.

SkyZeroZx committed:
docs: Add programmatic component input/output/directive binding

Expands the guide to explain how inputs, outputs, and directives can be provided when creating components programmatically

SkyZeroZx commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+
+## Providing inputs, outputs, and directives at creation
+
+The `createComponent` and `ViewContainerRef.createComponent` API supports binding inputs/outputs and attaching directives in one step. Instead of manually setting inputs and subscribing to outputs after creation, you can pass a `bindings` array with helpers like `inputBinding()`, `outputBinding()`, and `twoWayBinding()`, as well as a `directives` array for any directives to apply. This allows you ...
```
Updated, adds a comparison between both

SkyZeroZx commented on adev/src/content/guide/components/programmatic-rendering.md:
```diff
@@ -137,3 +137,106 @@ export class AdminSettings {
+
+### Popup attached to `document.body` with `createComponent` + `hostElement`
+
+Use this when you need to render outside the current view hierarchy (e.g., overlay to body). Lets you configure **bindings** directly.
```
A description about it is added
